packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1742,9 +1742,11 @@ class _TabBarState extends State<TabBar> {
         wrappedTabs[previousIndex] = _buildStyledTab(wrappedTabs[previousIndex], false, animation, _defaults);
       } else {
         // The user is dragging the TabBarView's PageView left or right.
-        final int tabIndex = _currentIndex!;
-        final Animation<double> centerAnimation = _DragAnimation(_controller!, tabIndex);
-        wrappedTabs[tabIndex] = _buildStyledTab(wrappedTabs[tabIndex], true, centerAnimation, _defaults);
+        if (_currentIndex! < widget.tabs.length) {
```

Comment:
I believe it's the developer responsibility to prevent the index to exceed the `tabs.length`, this check should be unnecessary

Reply:
the issue referenced refers to an error that in my opinion is caused by the implementation, not by flutter itself
see my [comment](https://github.com/flutter/flutter/issues/154484#issuecomment-2323448022)